echopype/echodata/echodata.py

```diff
@@ -133,6 +142,84 @@ def __read_converted(self, converted_raw_path: Optional["PathHint"]):
 
         self.converted_raw_path = converted_raw_path
 
+    def _set_tree(self, tree: DataTree):
+        self._tree = tree
+
+    @classmethod
+    def from_file(cls, converted_raw_path, storage_options, open_kwargs={}):
+        echodata = cls(
+            converted_raw_path=converted_raw_path,
+            storage_options=storage_options,
+            open_kwargs=open_kwargs,
+        )
+        echodata._check_path(converted_raw_path)
+        converted_raw_path = echodata._sanitize_path(converted_raw_path)
+        suffix = echodata._check_suffix(converted_raw_path)
+        tree = open_datatree(
+            converted_raw_path,
+            engine=XARRAY_ENGINE_MAP[suffix],
+            **echodata.open_kwargs,
+        )
+        # Change root name to top
+        tree.name = "Top-level"
+
+        echodata._set_tree(tree)
+
+        if isinstance(converted_raw_path, fsspec.FSMap):
+            # Convert fsmap to Path so it can be used
+            # for retrieving the path strings
+            converted_raw_path = Path(converted_raw_path.root)
+        echodata.converted_raw_path = converted_raw_path
+        echodata._load_tree()
+        return echodata
+
+    def _load_tree(self):
+        if self._tree is None:
+            raise ValueError("Datatree not found!")
+
+        for group, value in self.group_map.items():
+            # EK80 data may have a Beam_power group if both complex and power data exist.
+            ds = None
+            try:
+                if value["name"] == "Top-level":
+                    ds = self._tree.ds
+                else:
+                    ds = self._tree[value["name"]].ds
+            except ChildResolverError:
+                # Skips group not found errors for EK80 and ADCP
+                ...
+            if group == "top" and hasattr(ds, "keywords"):
+                self.sonar_model = ds.keywords.upper()  # type: ignore
+
+            if isinstance(ds, xr.Dataset):
+                setattr(self, group, ds)
```

Comment:
This is similar as _load_file for now to not break any current Echodata functionalities, though it grabs the dataset from the tree instead of loading it one by one.